### PR TITLE
Decompile func_801BBE70, add duplicates

### PR DESCRIPTION
**Pull request by sozud**

### CODE REVIEW

File: src/st/no3/377D4.c [hunk 1]
```diff
@@ -181,7 +198,24 @@ void func_801BBDC0(Entity* arg0) {
...
+        InitializeEntity(&D_80180B18);
+        arg0->zPriority = 0x29;
+        arg0->unk34 &= 0xF7FFFFFF;
+        arg0->animationFrame = arg0->subId + 0x22;
+        arg0->posX.i.hi = D_8018139C[arg0->subId << 1];
+        arg0->posY.i.hi = D_8018139E[arg0->subId << 1];
+        arg0->unk80.modeS16.unk0 = 5;
+    }
+    temp_v0 = arg0->unk80.modeS16.unk0 - 1;
+    arg0->unk80.modeS16.unk0 = temp_v0;
```
Comment (sonicdcer) [inline src/st/no3/377D4.c:214]:
```suggestion
arg0->unk80.modeS16.unk0--;
```

**Review by sonicdcer — CHANGES_REQUESTED** [review]

File: src/st/no3/377D4.c [hunk 2]
```diff
@@ -181,7 +198,24 @@ void func_801BBDC0(Entity* arg0) {
     }
 }
 
-INCLUDE_ASM("asm/st/no3/nonmatchings/377D4", func_801BBE70);
+void func_801BBE70(Entity* arg0) {
```
Comment (Xeeynamo) [inline src/st/no3/377D4.c:201]: `arg0` :arrow_right: `entity`

File: src/st/no3/377D4.c [hunk 3]
```diff
@@ -181,7 +198,24 @@ void func_801BBDC0(Entity* arg0) {
...
+    u16 temp_v0;
+
+    if (arg0->step == 0) {
+        InitializeEntity(&D_80180B18);
```
Comment (Xeeynamo) [inline src/st/no3/377D4.c:205]: Any parameter to `InitializeEntity` is an array of values. So `InitializeEntity(&D_80180B18)` becomes `InitializeEntity(D_80180B18)`(`&` removed) and `extern u16 D_80180B18;` becomes `extern u16 D_80180B18[];`

**Review by Xeeynamo — CHANGES_REQUESTED** [review]
Minor nitpick, but everything else looks very good!

**Comment by sozud** [comment]
Thanks, the comments should be addressed now.

**Pull request merged**